In [1]:
#imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import spacy
import tweepy

In [18]:
#read in twitter_archive data
df = pd.read_csv('twitter-archive-enhanced.csv')
df.head()

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
0  892420643555336193                    NaN                  NaN   
1  892177421306343426                    NaN                  NaN   
2  891815181378084864                    NaN                  NaN   
3  891689557279858688                    NaN                  NaN   
4  891327558926688256                    NaN                  NaN   

                   timestamp  \
0  2017-08-01 16:23:56 +0000   
1  2017-08-01 00:17:27 +0000   
2  2017-07-31 00:18:03 +0000   
3  2017-07-30 15:58:51 +0000   
4  2017-07-29 16:00:24 +0000   

                                              source  \
0  <a href="http://twitter.com/download/iphone" r...   
1  <a href="http://twitter.com/download/iphone" r...   
2  <a href="http://twitter.com/download/iphone" r...   
3  <a href="http://twitter.com/download/iphone" r...   
4  <a href="http://twitter.com/download/iphone" r...   

                                                text  retweeted_status_id  \
0  This is Phineas. He's a mystical boy. Only eve...                  NaN   
1  This is Tilly. She's just checking pup on you....                  NaN   
2  This is Archie. He is a rare Norwegian Pouncin...                  NaN   
3  This is Darla. She commenced a snooze mid meal...                  NaN   
4  This is Franklin. He would like you to stop ca...                  NaN   

   retweeted_status_user_id retweeted_status_timestamp  \
0                       NaN                        NaN   
1                       NaN                        NaN   
2                       NaN                        NaN   
3                       NaN                        NaN   
4                       NaN                        NaN   

                                       expanded_urls  rating_numerator  \
0  https://twitter.com/dog_rates/status/892420643...                13   
1  https://twitter.com/dog_rates/status/892177421...                13   
2  https://twitter.com/dog_rates/status/891815181...                12   
3  https://twitter.com/dog_rates/status/891689557...                13   
4  https://twitter.com/dog_rates/status/891327558...                12   

   rating_denominator      name doggo floofer pupper puppo  
0                  10   Phineas  None    None   None  None  
1                  10     Tilly  None    None   None  None  
2                  10    Archie  None    None   None  None  
3                  10     Darla  None    None   None  None  
4                  10  Franklin  None    None   None  None

In [46]:
#read in tweet_data
import tweepy

consumer_key = 'dt9P0F05cMwxz0VvcZ5ge5MJ4'
consumer_secret = 'OFI6CFxRdMs77rP7r6doflsY7sgD01wUFMjimN9KCaHZK2iyOc'
access_token = '943665104955961344-Pzk9xUglfts2sD2gOgUcUXdl32bDu6L'
access_secret = '71j24E9Ie9t9mz93vzfX8FH2EZbXi4kLETMRwP8HdmyOZ'

auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_secret)

api = tweepy.API(auth)

In [57]:
tweet = api.get_status(892420643555336193)
print(tweet.text)

This is Phineas. He's a mystical boy. Only ever appears in the hole of a donut. 13/10 https://t.co/MgUWQ76dJU


In [66]:
print(tweet.retweet_count)
print(tweet.favorite_count)


8199
37588
User(_api=<tweepy.api.API object at 0x7faecc2ad940>, _json={'id': 4196983835, 'id_str': '4196983835', 'name': 'WeRateDogs™', 'screen_name': 'dog_rates', 'location': '「 DM YOUR DOGS 」', 'description': 'Your Only Source For Professional Dog Ratings Instagram and Facebook ➪ WeRateDogs partnerships@weratedogs.com   ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀ cover photo: @TPaningning', 'url': 'https://t.co/N7sNNHSfPq', 'entities': {'url': {'urls': [{'url': 'https://t.co/N7sNNHSfPq', 'expanded_url': 'http://weratedogs.com', 'display_url': 'weratedogs.com', 'indices': [0, 23]}]}, 'description': {'urls': []}}, 'protected': False, 'followers_count': 7995664, 'friends_count': 12, 'listed_count': 6169, 'created_at': 'Sun Nov 15 21:41:29 +0000 2015', 'favourites_count': 141714, 'utc_offset': None, 'time_zone': None, 'geo_enabled': True, 'verified': True, 'statuses_count': 10094, 'lang': 'en', 'contributors_enabled': False, 'is_translator': False, 'is_translation_enabled': False, 'profile_background_color': '000000',

In [19]:
#print column names
df.columns

Index(['tweet_id', 'in_reply_to_status_id', 'in_reply_to_user_id', 'timestamp',
       'source', 'text', 'retweeted_status_id', 'retweeted_status_user_id',
       'retweeted_status_timestamp', 'expanded_urls', 'rating_numerator',
       'rating_denominator', 'name', 'doggo', 'floofer', 'pupper', 'puppo'],
      dtype='object')

In [21]:
#keep only certain columns
df = df[['tweet_id','timestamp','text','rating_numerator','rating_denominator','name', 'doggo', 'floofer', 'pupper', 'puppo']]
df.head()

tweet_id                  timestamp  \
0  892420643555336193  2017-08-01 16:23:56 +0000   
1  892177421306343426  2017-08-01 00:17:27 +0000   
2  891815181378084864  2017-07-31 00:18:03 +0000   
3  891689557279858688  2017-07-30 15:58:51 +0000   
4  891327558926688256  2017-07-29 16:00:24 +0000   

                                                text  rating_numerator  \
0  This is Phineas. He's a mystical boy. Only eve...                13   
1  This is Tilly. She's just checking pup on you....                13   
2  This is Archie. He is a rare Norwegian Pouncin...                12   
3  This is Darla. She commenced a snooze mid meal...                13   
4  This is Franklin. He would like you to stop ca...                12   

   rating_denominator      name doggo floofer pupper puppo  
0                  10   Phineas  None    None   None  None  
1                  10     Tilly  None    None   None  None  
2                  10    Archie  None    None   None  None  
3                  10     Darla  None    None   None  None  
4                  10  Franklin  None    None   None  None

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2356 entries, 0 to 2355
Data columns (total 10 columns):
tweet_id              2356 non-null int64
timestamp             2356 non-null object
text                  2356 non-null object
rating_numerator      2356 non-null int64
rating_denominator    2356 non-null int64
name                  2356 non-null object
doggo                 2356 non-null object
floofer               2356 non-null object
pupper                2356 non-null object
puppo                 2356 non-null object
dtypes: int64(3), object(7)
memory usage: 184.1+ KB


In [23]:
df.sample(6)

tweet_id                  timestamp  \
234   847617282490613760  2017-03-31 01:11:22 +0000   
476   816014286006976512  2017-01-02 20:12:21 +0000   
599   798682547630837760  2016-11-16 00:22:12 +0000   
60    880465832366813184  2017-06-29 16:39:47 +0000   
458   817908911860748288  2017-01-08 01:40:55 +0000   
1451  695794761660297217  2016-02-06 02:22:53 +0000   

                                                   text  rating_numerator  \
234   .@breaannanicolee PUPDATE: Cannon has a heart ...                13   
476   RT @dog_rates: This is Larry. He has no self c...                11   
599   RT @dog_rates: Here we see a rare pouched pupp...                 8   
60    This is Bella. She had her first beach experie...                12   
458   Looks like he went cross-eyed trying way too h...                12   
1451  This is Wyatt. His throne is modeled after him...                13   

      rating_denominator   name doggo floofer  pupper puppo  
234                   10   None  None    None    None  None  
476                   10  Larry  None    None    None  None  
599                   10   None  None    None  pupper  None  
60                    10  Bella  None    None    None  None  
458                   10   None  None    None    None  None  
1451                  10  Wyatt  None    None    None  None

In [22]:
df.describe()

tweet_id  rating_numerator  rating_denominator
count  2.356000e+03       2356.000000         2356.000000
mean   7.427716e+17         13.126486           10.455433
std    6.856705e+16         45.876648            6.745237
min    6.660209e+17          0.000000            0.000000
25%    6.783989e+17         10.000000           10.000000
50%    7.196279e+17         11.000000           10.000000
75%    7.993373e+17         12.000000           10.000000
max    8.924206e+17       1776.000000          170.000000

### tasks:
1. max rating denominator is 170. 
2. Standard deviation for rating denom 6.7. There could be other entries with > 10 values
3. max rating numerator is 1776. Seems very large and erroneous

In [39]:
max_count = len(df.rating_denominator.value_counts().values > 10)
print ("There are {} rating denominators greater than 10".format(max_count)) 

There are 18 rating denominators greater than 10


In [41]:
df.rating_numerator.value_counts()

12      558
11      464
10      461
13      351
9       158
8       102
7        55
14       54
5        37
6        32
3        19
4        17
1         9
2         9
420       2
0         2
15        2
75        2
80        1
20        1
24        1
26        1
44        1
50        1
60        1
165       1
84        1
88        1
144       1
182       1
143       1
666       1
960       1
1776      1
17        1
27        1
45        1
99        1
121       1
204       1
Name: rating_numerator, dtype: int64